### ISSUE TITLE
JSON on queue becomes string "System.Collections.Hashtable" in PowerShell

### ISSUE THREAD

**Issue by pacon-vib** [Azure/azure-functions-core-tools#2086]
The boilerplate PowerShell `run.ps1` that `func init` generates includes the following parameter declaration:

```pwsh
param([string] $QueueItem, $TriggerMetadata)
```

If you put a JSON string on the queue, then the value of `$QueueItem` becomes the string "System.Collections.Hashtable".

I changed the `param` block to cast `$QueueItem` to `[object]` instead of `[string]`, and now `$QueueItem` contains the decoded JSON object.

I don't know much about Azure Functions or PowerShell so I could be wrong, but I think the default should be something like `[object]` rather than `[string]`.

Please let me know if this ticket is in the wrong place.

**Comment by anthonychu** [comment]
@ggailey777 Do you know if there are any docs/tutorials that are using this template?

This is the same as what we currently have for TypeScript, we have the message annotated as a string but it could receive an object if the payload is JSON. @AnatoliB Do you have any thoughts?

**Comment by AnatoliB** [comment]
I think `[string]` in the PowerShell template is a mistake. It should be either removed or replaced with `[object]` (which is functionally equivalent).

From the PowerShell perspective, there is no bug here: the JSON content is automatically deserialized into a Hashtable, and then `[string]` converts it to a string by simply applying `.ToString()`, which uses the default `ToString` implementation and prints out the type name.

I have my doubts about the hidden JSON deserialization here. However, it's been there for long time and I'm sure there are many apps already relying on that, so I wouldn't change that now.

@anthonychu What currently happens in TypeScript when the message is annotated as a string but the payload is JSON? Is the payload automatically deserialized? What is the message value then? I would guess that TypeScript serializes the object back to JSON. Do you know?

**Comment by apawast** [comment]
@AnatoliB do you recommend we update the template here?

**Comment by AnatoliB** [comment]
@apawast Yes, I think so: we should either remove `[string]` (my preference) or replace it with `[object]`.
